tools/shared.py

```diff
@@ -1292,13 +1293,13 @@ def __getattr__(self, attr):
 
     def __setattr__(self, attr, value):
       if attr not in self.attrs:
-        import difflib
-        logging.warning('''Assigning a non-existent settings attribute "%s"''' % attr)
+        logging.error('Assigning a non-existent settings attribute "%s"' % attr)
         suggestions = ', '.join(difflib.get_close_matches(attr, list(self.attrs.keys())))
         if suggestions:
-          logging.warning(''' - did you mean one of %s?''' % suggestions)
-        logging.warning(''' - perhaps a typo in emcc's  -s X=Y  notation?''')
-        logging.warning(''' - (see src/settings.js for valid values)''')
+          logging.error(' - did you mean one of %s?' % suggestions)
+        logging.error(" - perhaps a typo in emcc's  -s X=Y  notation?")
```

Comment:
hmm this text is now out of date since we allow `-s X` as well, might be worth noting that here

Reply:
I think the pointer to settings.js should be enough since error message is already very long.  However, I should document the new syntax in settings.js for sure..

Reply:
sgtm